nbinteract/figure.py

```diff
@@ -0,0 +1,259 @@
+    'ylim', 'marker'
+])
+def _Scatter(x_fn, y_fn, *, options={}):
+    params = {
+        'marks': [{
+            'x': plotting._array_or_placeholder(x_fn),
+            'y': plotting._array_or_placeholder(y_fn),
+            'marker': plotting._get_option('marker'),
+        }]
+    }
+    fig = options.get('_fig', False) or _create_fig(options=options)
+    [scat] = plotting._create_marks(
+        fig=fig, marks=[bq.Scatter], options=options, params=params
+    )
+    plotting._add_marks(fig, [scat])
+
+    def x_fn_wrapped(**interact_params):
+        x_data = util.maybe_call(x_fn, interact_params, prefix='x')
+        scat.x = x_data
+
+    def y_fn_wrapped(**interact_params):
+        y_data = util.maybe_call(y_fn, interact_params, prefix='y')
+        scat.y = y_data
+
+    return [x_fn_wrapped, y_fn_wrapped]
+
+
+@plotting.use_options([
+    'title', 'aspect_ratio', 'animation_duration', 'xlabel', 'ylabel', 'xlim',
+    'ylim'
+])
+def _Line(x_fn, y_fn, *, options={}):
+    """
+    Internal Line function that generates the Line marker and the wrapper
+    functions for the line.
+    """
+    fig = options.get('_fig', False)
+    [line] = plotting._create_marks(fig=fig, marks=[bq.Lines], options=options)
+    plotting._add_marks(fig, [line])
+
+    def x_fn_wrapped(**interact_params):
+        x_data = util.maybe_call(x_fn, interact_params, prefix='x')
+        line.x = x_data
+
+    def y_fn_wrapped(**interact_params):
+        y_data = util.maybe_call(y_fn, interact_params, prefix='y')
+        line.y = y_data
+
+    return [x_fn_wrapped, y_fn_wrapped]
+
+
+class Plot(object):
+
+    NAME_TO_FN = {
+        'hist': _Hist,
+        'bar': _Bar,
+        'scatter': _Scatter,
+        'line': _Line
+    }
+
+    def __init__(self, options={}, widgets={}, functions={}, marks={}):
+        self.options = options
+        self.widgets = widgets
+        self.functions = functions
+        self.marks = marks.values()
+        self.figure = plotting._create_fig_with_options(options=options)
+        self.plot = plotting._create_fig_with_options(options=options)
+        self.create_plot()
+
+    def create_plot(self):
+        """
+        Generates the appropriate plot markers for each marker in the marks
+        dictionary and links them with the specified widgets.
+        """
+        for mark_features_dict in self.marks:
+            mark_name = mark_features_dict['marker']
+            mark_fn = Plot.NAME_TO_FN[mark_name]
+            # add in self.figure to options
+            mark_features_dict['options'] = tz.assoc(
+                mark_features_dict['options'], '_fig', self.figure
+            )
+            # removed mark from dictionary because it's not a param of
+            # the plotting function
+            no_marks_dict = self.minus_key('marker', mark_features_dict)
+            wrapped_fns = mark_fn(**no_marks_dict)
+            if mark_features_dict.get('x_fn'):
```

Comment:
Let's just use `if x_fn in mark_features_dict` since it's more Pythonic.

Reply:
Done!